example-scenegraph-sdk/source/tests/TestBasics.brs

```diff
@@ -55,4 +118,134 @@ function Test_Consent() as String
+    
+    print  ("Logging session after ad click: " + formatjson(mediaSession))
+    
+    m.mp.media.logAdEnd(mediaSession, {})
+    mediaSession.adContent = invalid
+    
+    print  ("Logging session after ad end: " + formatjson(mediaSession))
+    
+    adContent2 = mpConstants.adContent.build("ABC543", "VtM: B2 - Own the night")
+    adContent2.duration = 16000
+    adContent2.position = 0
+    adContent2.campaign = "VtM: Revival"
+    mediaSession.adContent = adContent2
+    m.mp.media.logAdStart(mediaSession, {})
+    
+    print  ("Logging session after ad start 2: " + formatjson(mediaSession))
+    
+    m.mp.media.logAdSkip(mediaSession, {})
+    mediaSession.adContent = invalid
+    
+    print  ("Logging session after ad skip: " + formatjson(mediaSession))
+    
+    m.mp.media.logAdBreakEnd(mediaSession, {})
+    mediaSession.adBreak = invalid
+    
+    print  ("Logging session after adBreak end: " + formatjson(mediaSession))
+        
+    m.mp.media.logQoS(mediaSession, 120, 3, 1232133, 46, {})
+    
+    print  ("Logging session after QOS: " + formatjson(mediaSession))
+    
+    m.mp.media.logSegmentEnd(mediaSession, customAttributes)
+    mediaSession.segment = invalid
+    
+    print  ("Logging session after segment end: " + formatjson(mediaSession))
+    
+    segment2 = mpConstants.Segment.build("Chapter 2", 1, 17500)
+    mediaSession.segment = segment2
+    m.mp.media.logSegmentStart(mediaSession, customAttributes)
+    
+    print  ("Logging session after segment start 2: " + formatjson(mediaSession))
+    
+    m.mp.media.logSeekStart(mediaSession, 1900, customAttributes)
+    
+    print  ("Logging session after seek start: " + formatjson(mediaSession))
+    
+    m.mp.media.logSeekEnd(mediaSession, 45600, {})
+    
+    print  ("Logging session after seek end: " + formatjson(mediaSession))
+    
+    m.mp.media.logSegmentSkip(mediaSession, customAttributes)
+    mediaSession.segment = invalid
+    
+    print  ("Logging session after segment skip: " + formatjson(mediaSession))
+    
+    m.mp.media.logBufferStart(mediaSession, 1200, 0.12, 45600, customAttributes)
+    
+    print  ("Logging session after buffer start: " + formatjson(mediaSession))
+    
+    m.mp.media.logBufferEnd(mediaSession, 563300, 1.0, 45600, {})
+    
+    print  ("Logging session after buffer end: " + formatjson(mediaSession))
+    
+    m.mp.media.logMediaContentEnd(mediaSession, customAttributes)
+    
+    print  ("Logging session after media content End: " + formatjson(mediaSession))
+    
+    m.mp.media.logMediaSessionEnd(mediaSession, customAttributes)
+    
+    print  ("Logging session after End: " + formatjson(mediaSession))
+    
+    return m.AssertEqual(formatjson(mediaSession), {"adbreak":null,"adcontent":null,"contentid":"ABC123","contenttype":"Video","currentplayheadposition":1900,"duration":1800000,"mediasessionid":"312c0508-744a-44bc-9285-f47359398e5d","segment":null,"streamtype":"LiveStream","title":"Space Pilot 3000"}
```

Comment:
rather than print, shouldn't each of them be m.AssertEqual?  it seems like what you're doing here is equivalent to JS's console.log(mediaSession), ,which is just logging, rather than expecting the value to be something.